quipucords/scanner/network/initial_processing.py

```diff
@@ -0,0 +1,373 @@
+
+        if value.get('rc', 0) and \
+           not getattr(processor, 'RETURN_CODE_ANY', False):
+            result[key] = 'Error: remote command returned {0}'.format(
+                value['stdout'])
+            continue
+
+        try:
+            processor_out = processor.process(value)
+        except Exception as ex:  # pylint: disable=broad-except
+            result[key] = 'Error: processor returned {0}'.format(str(ex))
+            continue
+
+        if isinstance(processor_out, Exception):
+            result[key] = 'Error: {0}'.format(' '.join(processor_out.args))
+            continue
+
+        result[key] = json.dumps(processor_out,
+                                 indent=' ', separators=(',', ':'))
+
+    return result
+
+
+class ProcessorMeta(abc.ABCMeta):
+    """Metaclass to automatically register Processors."""
+
+    def __init__(cls, name, bases, dct):
+        """Register cls in the PROCESSORS dictionary."""
+        if 'KEY' not in dct:
+            raise Exception('Processor {0} does not have a KEY'.format(name))
+
+        # Setting a falsey KEY means "yes, I did this on purpose, I
+        # really don't want this class registered."
+        if dct['KEY']:
+            PROCESSORS[dct['KEY']] = cls
+        super().__init__(name, bases, dct)
+
+
+# pylint: disable=too-few-public-methods
+class Processor(object, metaclass=ProcessorMeta):
+    """A class to process the output of an Ansible task."""
+
+    # Special fields:
+    #   KEY: a string. Should match the Ansible fact name that this
+    #     will process (required).
+    #   DEPS: a list of keys of other facts that this one depends
+    #     on. If one of the deps had an error, we will show a "missing
+    #     dependency" message rather than an error for this fact
+    #     (optional).
+    #   RETURN_CODE_ANY: if True, process() will pass results with
+    #     non-zero return code to this processor. If False, process()
+    #     will emit a default error message if the shell command has
+    #     non-zero return code (optional, defaults to False).
+    #   process(output): a static method. Process the output of the
+    #     task (required).
+
+    KEY = None
+
+    @staticmethod
+    def process(output):
+        """Process Ansible output.
+
+        :param output: an Ansible output dictionary. This typically
+          has members 'rc', 'stdout', and 'stdout_lines', but may
+          instead have a member 'results', which is an array of dicts
+          that each have 'rc', 'stdout', 'stdout_lines', and
+          'item'.
+
+        :returns: a Python object that represents output. Will return
+          an instance of Exception if it can't find the expected
+          information.
+        """
+        raise NotImplementedError()
+
+
+# #### Processors ####
+
+FIND_WARNING = 'find: WARNING: Hard link count is wrong for /proc: this may' \
+               ' be a bug in your filesystem driver.'
+
+
+class ProcessJbossEapRunningPaths(Processor):
```

Comment:
Maybe break these out into a `scanner/network/eap_processing/__init__.py`? and keep the common ones in here?

Or move this file to `scanner/network/processing/__init__.py` and create and `eap.py` with the eap specific ones?